### DISCUSSION TITLE
Only load select options once per request?

### DISCUSSION THREAD

**Discussion by jaxramus** [laravel/nova-issues#4375] [Q&A] [answered]
I have a Select field that looks like this:

    Select::make('Icon')->options($this->iconOptions())->searchable(),

Here is my `iconOptions` method:

    public function iconOptions()
    {
        $iconYaml = Yaml::parse(file_get_contents(resource_path('icons/icons.yml')));
        $iconOptions = [];

        foreach ($iconYaml as $key => $value) {
            if (strlen($key) > 1 && in_array('solid', $value['styles'])) {
                $iconOptions["fa-solid fa-$key"] = $value['label'];
            }
            else if (in_array('brands', $value['styles'])) {
                $iconOptions["fa-brands fa-$key"] = $value['label'];
            }
        }

        return $iconOptions;
    }

This method parses the Font Awesome icon YAML file to return an array of icon classes.

My problem is that, when viewing things like the resource index, it is hitting the `iconOptions` method for every row. Since the `icons.yml` file is quite large, this slows down the request a lot.

How do I get Nova to only load hit the `iconOptions` method once per request?

**Comment by jaxramus** [comment]
Ended up just doing it with caching.

**Comment by crynobone** [comment, marked as answer]
You can also use closure so it only resolve when needed 

```php
Select::make('Icon')->options(function () { return  $this->iconOptions(); })->searchable()
```
